### ISSUE TITLE
Significantly reduced validation accuracy when switching from alpha0 to beta0/1

### ISSUE THREAD

**Issue by taotsetung** [tensorflow/tensorflow#30431]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):  Code is strongly based on: [Stock example](https://www.tensorflow.org/beta/tutorials/images/transfer_learning ), but uses kaggle aerial-cactus-identification dataset. 
- OS Platform and Distribution: Google Colab
- TensorFlow installed from: binary via `pip install tensorflow-gpu==2.0.0-beta0`
- TensorFlow version: 2.0.0-beta0
- Python version: 3.6.7
- CUDA/cuDNN version: N/A (Google Colab as of 05.07.2019)
- GPU model and memory:  N/A (Google Colab as of 05.07.2019)

**Describe the current behavior**
With beta 0 and 1 the training/validation history is as follows:
![Beta 0/1](https://puu.sh/DOY9H/8ec1a1a7f5.png)
This is a very low validation accuracy. This can of course happen, even with code based on an example, the issue is that this only appears with beta0 and beta1 builds, not with alpha0 (see below) when using the exactly same code and training data.

**Describe the expected behavior**
With alpha0 and the same code the history looks as follows:
![Alpha 0](https://puu.sh/DOYss/b14cc7f979.png)

An upgrade of the tensorflow version should not affect the resulting accuracy in such a manner.

**Code to reproduce the issue**
1. Open this [Google Colab](https://colab.research.google.com/drive/14oaC63n1n3yymRyB90uMx3GNvzt-rGMT)
2. Run the code with Alpha0
![Choose Version](https://puu.sh/DOYdz/3c3efe344c.png)
3. Make note of the training history plot.
4. Restart the runtime.
5. Run the code with Beta0 or Beta1
![Choose Version](https://puu.sh/DOYeP/6a24c7c681.png)
6. Make note of the training history plot.
7. Observe that with no change to the code but using the beta0 instead of alpha0 the validation accuracy goes down from > 95% to < 90%, with beta1 even to < 80%

**Other info / logs**
My best guess is that changes have been made to the pretrained MobileNetV2 or to the Adam optimizer, otherwise the drastic loss in accuracy is hard to explin.

**Comment by ravikyram** [comment]
I have tried on colab with TF version 2.0 beta 1 and alpha 0 and was able to reproduce the issue.Thanks!

**Comment by karmel** [comment]
Could be related to the changes in trainable flags between alpha and beta; @tanzhenyu , can you take a look?

**Comment by lukaszgrad** [comment]
I encountered the same issue with beta1 version. I'm also using tf.keras.Model.fit() (but with distributed Dataset)

Hope you guys can get this bug fixed, since people experimenting with tf 2.0 will get unsatisfactory results from their models. 

Cheers!

**Comment by tanzhenyu** [comment]
I was able to reproduce this in colab. Will look into it.

**Comment by AbrahamSanders** [comment]
Hi,
I also encountered this issue in a colab from a course demonstrating transfer learning with MobileNetV2.
I ran the same exact code in alpha0 and then in beta1, and am seeing a reduction of 13.6% in validation accuracy in beta1.
[Colab for Alpha0](https://colab.research.google.com/drive/1KszPPoBVwBoAGwgoSMtUjuLZMYRXzvWV#scrollTo=6XOGP0tExZvv)
[Colab for Beta1](https://colab.research.google.com/drive/1-s53y_oAMaa1VCZl7JqpzNCAoVumk1Qj#scrollTo=6XOGP0tExZvv)

I hope this aids in your diagnosis of the issue.

**Comment by fchollet** [comment]
In between these 2 version, we have changed the semantics of setting `batch_norm.trainable = False` on a `BatchNormalization` layer.

Old behavior: freeze weights, freeze updates, use training-time behavior during training (i.e. use the batch statistics are used for training-time normalization).

New behavior: freeze weights, freeze updates, use inference-time behavior during training (i.e. use the frozen batch statistics gamma and beta).

In general we expect the new behavior to work better for fine-tuning use cases. But in any case, it is *different*, so it's expected that you'd see different results.

Note that you can always manually specify what behavior you want by explicitly passing the `training` argument in `call` for your BatchNormalization layers when building the model you use for training (and changing it when you use a different model that will be for evaluation and prediction only).

**Comment by AbrahamSanders** [comment]
Thanks for the clarification @fchollet, indeed I see the documentation of the change in the comments of the [BatchNormalization](https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/layers/normalization_v2.py#L26-L65) class. 

So, since both @taotsetung and I observed this behavior with the MobileNetV2 model provided by tf.keras.applications.MobileNetV2, I presume this model was trained before the change (the weights throughout the model would have been updated based on the gradients of activations normalized using the training-time batch statistics and now don't "recognize" the activations normalized using the new behavior)? Could this explain the drastic drop in validation accuracy we observed and could the solution simply be to re-train the entire base model provided by tf.keras.applications.MobileNetV2 using the new behavior?

I hope I am understanding this correctly, my apologies if I am way off.

**Comment by AbrahamSanders** [comment]
Never mind, I realized just after posting that my suggestion made no sense since the behavior when trainable = True did not change. Re-training the entire model (all layers are trainable) would not lead to any different results.

**Comment by jvishnuvardhan** [comment]
I can still reproduce this issue in `!pip install tf-nightly-gpu-2.0-preview`. Please take a look at the [gist](
https://colab.sandbox.google.com/gist/jvishnuvardhan/62ae11bacb15ff883d337f9139de04b9/image-classification.ipynb). Thanks!
